bindings/wasm/src/did/wasm_resolved_document.rs

```diff
@@ -17,14 +17,25 @@ use wasm_bindgen::prelude::*;
 #[derive(Clone, Debug)]
 pub struct WasmResolvedDocument(pub(crate) ResolvedIotaDocument);
 
-// Workaround for Typescript type annotations on async function returns.
 #[wasm_bindgen]
 extern "C" {
+  // Workaround for Typescript type annotations on async function returns.
   #[wasm_bindgen(typescript_type = "Promise<ResolvedDocument>")]
   pub type PromiseResolvedDocument;
 
   #[wasm_bindgen(typescript_type = "Promise<Array<ResolvedDocument>>")]
   pub type PromiseArrayResolvedDocument;
+
+  // Workaround for (current) lack of array support in wasm-bindgen
+  #[wasm_bindgen(typescript_type = "Array<ResolvedDocument>")]
+  pub type ArrayResolvedDocument;
+
+  // Workaround for (current) lack of generics in wasm-bindgen
+  #[wasm_bindgen(typescript_type = "Document | ResolvedDocument")]
+  pub type DocumentOrResolvedDocument;
+
+  #[wasm_bindgen(typescript_type = "Array<Document> | Array<ResolvedDocument>")]
+  pub type ArrayDocumentOrArrayResolvedDocument;
```

Comment:
Optional.

```suggestion
  #[wasm_bindgen(typescript_type = "Array<Document> | Array<ResolvedDocument>")]
  pub type ArrayDocumentOrResolvedDocument;
```

Reply:
I considered `ArrayDocumentOrResolvedDocument`, but thought  the name could be interpreted as `Array<Document | ResolvedDocument>` which is not what we want.